core-bundle/src/Routing/ResponseContext/ResponseContext.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Routing\ResponseContext;
+
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\ResponseHeaderBag;
+
+class ResponseContext implements ResponseContextInterface
+{
+    /**
+     * @var ResponseHeaderBag|null
+     */
+    private $headerBag;
+
+    public function getHeaderBag(): ResponseHeaderBag
+    {
+        if (null === $this->headerBag) {
+            $this->headerBag = new ResponseHeaderBag();
```

Comment:
we must make sure this `ResponseHeaderBag` is empty so it only contains what is really set. The default class is not empty!

Reply:
Indeed but that's a problem. That's impossible. There's always at least `Date`. It's forced on the `ResponseHeaderBag`.

Reply:
If we take the default `HeaderBag` though, we don't have the cache control shortcuts.

Reply:
I have fixed that by having our own `PartialResponseHeaderBag` in 5e69b26d2dc6c46aee4d2eb0800a84cb0c53f15e. I'll mark this discussion as resolved then, feel free to create a new one.